### PR TITLE
WIP: fix documentation build with mike

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

As we only have a single documentation version for now, we need to set the default version to point to.

## Proposal

- [x] fix local mike configuration for development
- [x] set default documentation in the CI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -434,12 +434,17 @@ jobs:
             # - DOCS_VERSION: 1.1 (for git tag v1.1.2)
             # - DOCS_ALIAS: latest
             # or
-            # - DOCS_VERSION: master (for master branch)
+            # - DOCS_VERSION: main (for main branch)
             # - No DOCS_ALIAS
             DOCS_VERSION=$([[ -z "$CIRCLE_TAG" ]] && echo $CIRCLE_BRANCH || echo ${CIRCLE_TAG} | sed 's/^v\([0-9]\.[0-9]*\)\..*/\1/')
```
Comment (wilbrdt) [inline .circleci/config.yml:439]: I find that a documentation version named `dev` less confusing than having a `main`. WDYT?
```suggestion
            # - DOCS_VERSION: dev (for main branch)
            # - No DOCS_ALIAS
            DOCS_VERSION=$([[ -z "$CIRCLE_TAG" ]] && echo "dev" || echo ${CIRCLE_TAG} | sed 's/^v\([0-9]\.[0-9]*\)\..*/\1/')
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -434,12 +434,17 @@ jobs:
...
             DOCS_ALIAS=$([[ -z "$CIRCLE_TAG" ]] && echo "" || echo "latest")
             echo "DOCS_VERSION: ${DOCS_VERSION}"
             echo "DOCS_ALIAS: ${DOCS_ALIAS}"
+            # Build docs
+            ~/.local/bin/mike deploy ${DOCS_VERSION} ${DOCS_ALIAS}
+            # Set default doc to point to
+            ~/.local/bin/mike set-default main
```
Comment (wilbrdt) [inline .circleci/config.yml:446]: If my previous comment is accepted:
```suggestion
            ~/.local/bin/mike set-default dev
```

Comment (wilbrdt) [reply .circleci/config.yml:446]: This would change on the first release of Warren right? Pointing to the latest tag

File: Makefile [hunk 3]
```diff
@@ -366,20 +366,20 @@ test-app: run-app
 
 ## -- Docs
 docs-build: ## build documentation site
-	@$(MKDOCS) build
+	$(MKDOCS) build
 .PHONY: docs-build
 
-docs-deploy: ## deploy documentation site
+docs-deploy: ## build documentation site for all versions
```
Comment (wilbrdt) [inline Makefile:372]:
```suggestion
docs-deploy: ## build and deploy documentation site for all versions
```

**Review by wilbrdt — APPROVED** [review]
LGTM! :tada: 
Just a minor suggestion

File: .circleci/config.yml (hunk 1 above)
Comment (jmaupetit) [reply .circleci/config.yml:439]: In this case, what about `development` instead?

File: .circleci/config.yml (hunk 2 above)
Comment (jmaupetit) [reply .circleci/config.yml:446]: Yup.

**Pull request merged**